packages/sns/lib/sns/AbstractSnsSqsConsumer.ts

```diff
@@ -85,6 +85,8 @@ export abstract class AbstractSnsSqsConsumer<
         this.creationConfig.queue,
         this.creationConfig.topic,
         this.subscriptionConfig,
+        undefined,
+        this.locatorConfig,
```

Comment:
TS is complaining now that `initSnsSqs().queueUrl` is string|undefined.

Reply:
@kibertoad I solved this by adding `if` to check if queue url is present. From the code it looks it should be mandatory parameter if `subscriptionArn` is provided, we must provide `queueUrl` too. What do you think?

Reply:
isn't there branching after the check you added, where we skip the queueUrl-related logic if it's not present?

we should support passing queueName instead of queueUrl, no? or we always resolve url based on name later?

Reply:
There is branch with check, but it didn't guarantee that `queueUrl` is present after that:
```ts
let queueName: string
  if ((locatorConfig as SQSQueueLocatorType).queueUrl) {
    const splitUrl = (locatorConfig as SQSQueueLocatorType).queueUrl.split('/')
    queueName = splitUrl[splitUrl.length - 1]
  } else {
    queueName = creationConfig!.queue.QueueName!
  }

  return {
    subscriptionArn: locatorConfig.subscriptionArn,
    topicArn: subscriptionTopicArn,
    queueUrl: locatorConfig.queueUrl,
    queueName,
  }
```
From this code it looks that queueUrl is mandatory.

Reply:
I don't see it. if queueUrl is not present here, we will just return queueUrl: undefined

Reply:
Yep, but `queueUrl` cannot be null. It's required in `AbstractSnsSqsConsumer`, see line: 104